mautic/mautic
# Email report graph table: Most clicked email links

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds new report graph table with most clicked email links. The table is available in **Emails** and **Emails Sent** datasources.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create an email with links and send it to your contacts
3. Generate some clicks (open email and click link while not logged into Mautic in the same browser session)
4. Add new **Report** with **Email** datasource
5. Select some columns in the **Data** tab
![image](https://user-images.githubusercontent.com/8580942/224698186-75f114a5-e263-4d18-96d1-addae5cf5a22.png)
6. Add **Most clicked links** in the **Graph** Tab
![image](https://user-images.githubusercontent.com/8580942/224698373-f3e7b60e-d665-4190-a8d0-ff73d43b40ba.png)
7. Click **Save and Close** to view report
8. Verify the **Most clicked links** table below the standard data table
![image](https://user-images.githubusercontent.com/8580942/224697649-b13262b7-6a1f-4b5b-8556-5181c40d42a8.png)
9. Repeat steps 4-8 for **Emails Sent** datasource
![image](https://user-images.githubusercontent.com/8580942/224698804-4b605790-ad44-4322-bb3a-0fb5a76aeaa9.png)
